packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.test.ts

```diff
@@ -227,6 +229,238 @@ describe('GraphQLSitemapService', () => {
+                    hasNext: false,
+                  },
+                  results: [
+                    {
+                      path: '/y1',
+                    },
+                    {
+                      path: '/x1/x2',
+                    },
+                  ],
+                },
+              },
+            },
+          },
+        });
+
+      const service = new GraphQLSitemapService({
+        endpoint,
+        apiKey,
+        sites: multipleSites,
+      });
+      const sitemap = await service.fetchSSGSitemap([lang]);
+
+      // expect(getSiteRewriteStub.called).to.equal(true);
+
+      expect(sitemap).to.deep.equal([
+        {
+          params: {
+            path: ['_site_site1'],
+          },
+          locale: lang,
+        },
+        {
+          params: {
+            path: ['_site_site1', 'x1'],
+          },
+          locale: lang,
+        },
+        {
+          params: {
+            path: ['_site_site1', 'y1', 'y2', 'y3', 'y4'],
+          },
+          locale: lang,
+        },
+        {
+          params: {
+            path: ['_site_site1', 'y1', 'y2'],
+          },
+          locale: lang,
+        },
+        {
+          params: {
+            path: ['_site_site2', 'y1'],
+          },
+          locale: lang,
+        },
+        {
+          params: {
+            path: ['_site_site2', 'x1', 'x2'],
+          },
+          locale: lang,
+        },
+      ]);
+      return expect(nock.isDone()).to.be.true;
+    });
+
+    it('should return aggregated paths for multiple sites and personalized sites', async () => {
+      const multipleSites = ['site1', 'site2'];
+      const lang = 'ua';
+
+      nock(endpoint)
+        .post('/', /PersonalizeSitemapQuery/gi)
+        .reply(200, sitemapPersonalizeQueryResult);
+
+      nock(endpoint)
+        .persist()
+        .post('/', (body) => {
+          return body.variables.siteName === multipleSites[1];
+        })
+        .reply(200, {
+          data: {
+            site: {
+              siteInfo: {
+                routes: {
+                  total: 4,
+                  pageInfo: {
+                    hasNext: false,
+                  },
+                  results: [
+                    {
+                      path: '/',
+                    },
+                    {
+                      path: '/x1',
+                    },
+                    {
+                      path: '/y1/y2/y3/y4',
+                    },
+                    {
+                      path: '/y1/y2',
+                    },
+                  ],
+                },
+              },
+            },
+          },
+        });
+
+      const service = new GraphQLSitemapService({
+        endpoint,
+        apiKey,
+        sites: multipleSites,
+        includePersonalizedRoutes: true,
+      });
+      const sitemap = await service.fetchSSGSitemap([lang]);
+
+      expect(sitemap).to.deep.equal([
+        {
+          params: {
+            path: [''],
```

Comment:
paths for site1 should have a site1 prefix applied
when multisite is enabled, all sites will have a prefix. in case of personalization, two prefixes will be present